### ISSUE TITLE
Hints are not saved correctly $^H refcounted_he_hek

### ISSUE THREAD

**Issue by atoomic** [CpanelInc/perl-compiler#44]
This test is segfaulting on branch hv_stash with HEAD=73e9aeb96a79ef6127bb88a908f60dab7c7c47ad

This is an extract of op/sub.t

```perl
#!./perl

use feature 'lexical_subs'; 
no warnings 'experimental::lexical_subs';

my sub not_constantm () { 42 }

sub do_caller { my @caller = caller(1) }

sub x { do_caller() }

{
    sub re::regmust{}
    bless \&re::regmust; # force DESTROY
    DESTROY { print qq{DESTROYED\n} }
    require re;
    print qq{loaded re\n};
    x();
}
```

a simple gdb session show a SEGV there

```
(gdb) run
Starting program: /root/workspace/bc/t/v5.24.1/t/op/sub.bin
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
DESTROYED
loaded re

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff7880425 in Perl_refcounted_he_chain_2hv (chain=0x6803e0, flags=0) at hv.c:3163
3163		U32 hash = HEK_HASH(chain->refcounted_he_hek);
(gdb) bt
#0  0x00007ffff7880425 in Perl_refcounted_he_chain_2hv (chain=0x6803e0, flags=0) at hv.c:3163
#1  0x00007ffff78dd365 in Perl_pp_caller () at pp_ctl.c:1951
#2  0x00007ffff7883887 in Perl_runops_standard () at run.c:41
#3  0x00007ffff77c2c92 in S_run_body (oldscope=1) at perl.c:2485
#4  0x00007ffff77c278b in perl_run (my_perl=0x65c8b8 <BC_unthreaded_interpreter>) at perl.c:2408
#5  0x0000000000413d6c in main (argc=1, argv=0x7fffffffdd08, env=0x7fffffffdd18) at /root/workspace/bc/t/v5.24.1/t/op/sub.c:14104
```

using an uncompiled perl version we can notice that we should have the HEK for feature_lexsubs which is saved by feature.pm
```
$1 = (const struct refcounted_he *) 0x62ca90
(gdb) p	chain->refcounted_he_hek
$2 = (HEK *) 0x630948
(gdb) p *
$3 = {
  hek_hash = 1107553097,
  hek_len = 15,
  hek_key = 	"f"
}
(gdb) p	(char*)	chain->refcounted_he_hek->hek_key
$4 = 0x630950 "feature_lexsubs"
```

```
(gdb) p	*chain
$5 = {
  refcounted_he_next = 0x62ca30,
  refcounted_he_hek = 0x630948,
  refcounted_he_val = {
    refcounted_he_u_iv = 1,
    refcounted_he_u_uv = 1,
    refcounted_he_u_len	= 1,
    refcounted_he_u_ptr	= 0x1
  },
  refcounted_he_refcnt = 8,
  refcounted_he_data = 	   " "
}
```

**Comment by toddr** [comment]
bctestc/44.Hints-refcounted-he.t

**Comment by atoomic** [comment]
fixed with commit 2aa68f39e794b0e39afbe1b48b253414a4965192
